ipaserver/plugins/service.py

```diff
@@ -201,6 +201,28 @@ def validate_realm(ugettext, principal):
         raise errors.RealmMismatch()
 
 
+def validate_auth_indicator(entry):
+    new_value = entry.get('krbprincipalauthind', None)
+    if not new_value:
+        return
+    # The following services are considered internal IPA services
+    # and shouldn't be allowed to have auth indicators.
+    # https://pagure.io/freeipa/issue/8206
+    pkey = api.Object['service'].get_primary_key_from_dn(entry.dn)
+    principal = kerberos.Principal(pkey)
+    servers = (s['cn'][0] for s in api.Command.server_find()['result'])
```

Comment:
I think this has a potential for a performance issue. Instead of doing blind search for all IPA servers, please do use per-name search `ipa server-find --name={name}`.